Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift

```diff
@@ -425,7 +446,7 @@ extension ProjectDescription.Target {
         targetToModuleMap: [String: ModuleMap],
         addPlatformSuffix: Bool
     ) throws -> Self? {
-        guard target.type == .regular else {
+        guard target.type == .regular || target.type == .system else {
```

Comment:
Does a `.system` target require all the mapping of a `regular` target as well? If not maybe it makes sense to handle it separately?

Reply:
@danyf90 I think you're right, the `.system` target shouldn't support sources, resources, dependencies, and c/cxx/swift settings. 

I updated the code to skip the specific parsing steps based on the preconditions here:
https://github.com/apple/swift-package-manager/blob/37ace19c013245c45ef3d9e9ef2f5e00516ed6b9/Sources/PackageDescription/Target.swift#L275

The update would skip headers/sources/dependencies for `.system` targets. I'm not sure if there's a need for more granular settings processing to strip c/cxx/swift settings though.